deploy/examples/secure-query-with-keycloak-sidecar-values.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: v1
+kind: Jaeger
+  metadata:
+    name: jaeger
+    namespace: default
+  spec:
+    ingress:
+      enabled: true
+      security: "oauth-proxy"
+    query:
+      oauthProxy:
+        options:
+          #        Client ID for OpenID server
+          client-id: jaeger-client
+          #        Client secret for OpenID server.
+          #        Confidential access type is for server-side clients that need to perform a browser login and require a client
+          #        secret when they turn an access code into an access token, (see Access Token Request in the OAuth 2.0 spec for
+          #        more details). This type should be used for server-side applications.
+          #        see https://www.keycloak.org/docs/6.0/server_admin/
+          client-secret: 12345678-9abc-def0-1234-56789abcdef0
+          #        URL for OpenID autoconfiguration
+          discovery-url: https://keycloak.example.com/auth/realms/jaeger-realm
+          #        Enables a default denial on all requests, you have to explicitly say what is permitted (recommended)
+          enable-default-deny: true
+          #        The interface definition you wish the proxy to listen, all interfaces is specified as ':<port>', unix sockets as unix://<REL_PATH>|</ABS PATH>
+          listen: ":9091"
+          #        Preserve the host header of the proxied request in the upstream request
+          preserve-host: true
+          #        The redirection url, essentially the site url, note: /oauth/callback is added at the end
+          redirection-url: "https://jaeger-gatekeeper.example.com"
+          #        Role restrictions for URL. User must have specified role in order to access.
+          resources: "uri:/admin/*|white-listed=true"
```

Comment:
Have you tested this configuration? I'm not sure this will work. We don't have a `/admin` in the Jaeger UI.

Reply:
My bad... The `/admin` does not belong there.

Reply:
I corrected this but found another problem, especially with the `resources` option. The Gatekeeper accepts this option multiple times so you can define it for different subpages:
```
--enable-default-deny=true \
--resources="uri=/admin*|roles=test1,test2" \
--resources="uri=/backend*|roles=test1" \
```
Our current implementation with translation from yaml options to args only accepts this once and other notations (map, multiple uri's sperated with pipes) do not work properly.

Reply:
We enabled adding volumeMounts to the oauth-proxy to configure it via a configuration file placed in a secret